File: app/src/main/java/com/team2052/frckrawler/database/metric/MetricValue.java
package com.team2052.frckrawler.database.metric;

import com.google.gson.JsonElement;
import com.team2052.frckrawler.db.Metric;

public class MetricValue {
    private Metric metric;
    private JsonElement value;

    public MetricValue(Metric metric, JsonElement value) {
        if (metric == null)
            throw new IllegalStateException("Metric cannot be null");
        this.metric = metric;
        this.value = value;
    }

    public Metric getMetric() {
        return metric;
    }

    public JsonElement getValue() {
        return value;
    }

}

File: core/metrics/src/main/java/com/team2052/frckrawler/core/metrics/data/MetricValue.kt
package com.team2052.frckrawler.core.metrics.data

import com.google.gson.JsonElement
import com.team2052.frckrawler.core.common.MetricHelper
import com.team2052.frckrawler.core.common.v3.JSON
import com.team2052.frckrawler.core.data.models.Metric

data class MetricValue(val metric: Metric, val value: JsonElement?) {
    val metricType: Int
        @MetricHelper.MetricType
        get() = metric.type

    fun valueAsString(): String {
        return if (value == null) "" else JSON.getGson().toJson(value)
    }
}